dev/WindowsAppRuntime_DLL/dllmain.cpp

```diff
@@ -6,6 +6,10 @@
 #include <MddDetourPackageGraph.h>
 #include <urfw.h>
 
+// Including this file once per binary will automatically opt WIL error handling macros into calling RoOriginateError when they
+// begin logging a new error.  This greatly improves the debuggability of errors that propagate before a failfast.
+#include <wil/result_originate.h>
```

Comment:
Nice.

Please also add this to our other DllMain's

* dev\DynamicDependencyDataStore\DynamicDependency.DataStore\winmain.cpp
* dev\DynamicDependencyLifetimeManager\DynamicDependencyLifetimeManager\winmain.cpp
* dev\PushNotifications\PushNotificationsLongRunningTask\winmain.cpp
* dev\WindowsAppRuntime_BootstrapDLL\dllmain.cpp

P.S. DllMain should end with

```
    // Give WIL a crack at DLLMain processing
    // See DLLMain() in wil/result_macros.h for why
    wil::DLLMain(hmodule, reason, reserved);

    return TRUE;
}
```

? These don't and could use the addition

* dev\DynamicDependencyDataStore\DynamicDependency.DataStore\winmain.cpp
* dev\DynamicDependencyLifetimeManager\DynamicDependencyLifetimeManager\winmain.cpp
* dev\PushNotifications\PushNotificationsLongRunningTask\winmain.cpp

Reply:
This is great - anything to improve error reporting and actionability for devs is huge.  @dmachaj how about adding this to other feeders as well, like WinUI?

Reply:
This change will be helpful for any projects that use WIL error macros widely.  WinUI is not currently on that list because they have their own set of error-propagating macros.  If there are any other projects contributing to WinAppSDK that use WIL heavily then it would be great to add coverage.